logic.py

```diff
@@ -315,19 +317,19 @@ def scored_proposals():
     batch_ids = {v.proposal:v.batch_id for v in votes}
     """
     for v in votes:
-        scores[v.proposal].extend(v.scores.values())
```

Comment:
Same here. Extend takes an iterable so I don't think you need to switch to list.